blog/_posts/2017-10-12-gsoc-NeuralNetDiffEq.md

```diff
@@ -0,0 +1,191 @@
+
+## Research aspect of the project and the challenge
+The research paper we referred to on the topic is quite old and understanding the examples as well as explanations was quite challenging. Not much research has been done on using NNs for this purpose and thus we were not able to get much help from the research papers related to the topic.
+The initial task was to read and understand the mathematics behind solving differential equations. Also the computational methods used to solve differential equations on computers are quite different from the ones we use on paper so it took quite some time to get familiar with them. The structure and type of NN to be used so that the solver advantages (parallelism in time) are retained without compromising the performance was a research subdomain as well was a challenge.
+
+
+After implementing the solver for ODEs (Ordinary Differential Equations) and systems of ODEs, the difficult part was to make the NN converge for the systems of ODEs on longer time domains.
+As there are a lot of factors involved in neural networks, like hidden layer width, number of hidden neurons, activations, weights etc., I relied on my machine learning background as well as the help from my mentors to experiment with most of the feasible settings of NN hyper-parameters and recording the accuracy of convergence and performance of the solver.
+Making the NNs converge for systems of ODEs is not as easy as it seems and took up most of the time for experimentation and tuning. Predicting the system of DEs solution with larger domain is still a challenge which needs to be worked upon.
+
+## Implementation and Work
+The implementation involved integration of mathematical and machine learning aspects to build a neural net solver for ODEs. The [DiffEqBase library](https://github.com/JuliaDiffEq/DiffEqBase.jl) is used as a base to extend the algorithm and solver interface while the neural network was developed using the [Knet.jl library](https://github.com/denizyuret/Knet.jl). The work done till now can be seen on the [NeuralNetDiffEq.jl github repository](https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl), primarily in [this branch](https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl/tree/SingleNN_Approach).
+This work involves implementing a Neural Network solver for ODEs with customized interpolation based on NN prediction. 
+
+### How does it work?
+We construct a trial solution for our differential equation in terms of the NN output which should also satisfy the DE boundary conditions.
+We define a loss function for the neural net which is the difference between the derivative of the neural net solution and the true derivative defined by the ODE.
```

Comment:
make that "the derivative of the neural net solution with regards to its input"

Add after:

"This is an unusual loss function, in that in includes the gradient of the network itself. It is almost unseen elsewhere in other ML applications"